applications/asset_tracker_v2/doc/cloud_module.rst

```diff
@@ -0,0 +1,195 @@
+| `nRF Cloud`_     |    `MQTT`_                    |
+|                  +-------------------------------+
+|                  |    `TCP`_                     |
+|                  +-------------------------------+
+|                  |    `TLS`_                     |
+|                  +-------------------------------+
+|                  |    :ref:`FOTA <nrf9160_fota>` |
+|                  +-------------------------------+
+|                  |    :ref:`lib_nrf_cloud_agps`  |
+|                  +-------------------------------+
+|                  |    :ref:`lib_nrf_cloud_pgps`  |
++------------------+-------------------------------+
+
+.. _nrfcloud_agps_pgps:
+
+nRF Cloud A-GPS and P-GPS
+=========================
+
+When the cloud module is configured to communicate with `AWS IoT Core`_ or `Azure IoT Hub`_, it supports processing of received A-GPS and P-GPS data using the :ref:`lib_nrf_cloud_agps` and :ref:`lib_nrf_cloud_pgps` libraries.
+This enables the cloud to fetch A-GPS and P-GPS data from `nRF Cloud`_ using REST calls and relay the data to the nRF9160 SiP using the pre-established connection to `AWS IoT Core`_ or `Azure IoT Hub`_.
+Maintaining multiple cloud connections at the same time is in general discouraged seen from a data and energy consumption standpoint.
+This is mainly due to the extra overhead associated with establishing a secure connection, which typically consists of multiple kB of exchanged data with the cloud service.
+When configuring the application to communicate with nRF Cloud, A-GPS and P-GPS data are received directly from the service, in contrast to the AWS IoT and Azure IoT Hub implementations.
+
+FOTA
+====
+
+The client libraries supported by the cloud wrapper API all implement their own version of :ref:`FOTA <nrf9160_fota>`.
+This enables the cloud to issue FOTA updates while the device is in field to update the application and modem firmware.
+For additional documentation on the various FOTA implementations, refer to the respective client library documentation linked to in :ref:`Integration layers <integration_layers>`.
+
+Connection Awareness
+====================
+
+The cloud module implements connection awareness by maintaing an internal state that is based on
+events from the modem module and callbacks from the :ref:`Cloud wrapper API <api_cloud_wrapper>`.
+
+If at any point the module is disconnected, it will try to reconnect given that an LTE connection is still valid.
+The number of reconnection attempts can be adjusted by configuring the :ref:`CONFIG_CLOUD_CONNECT_RETRIES <CONFIG_CLOUD_CONNECT_RETRIES>` option.
+Reconnection is implemented with a binary backoff based on the following lookup table:
+
+.. code-block:: c
+
+   static struct cloud_backoff_delay_lookup backoff_delay[] = {
+      { 32 }, { 64 }, { 128 }, { 256 }, { 512 },
+      { 2048 }, { 4096 }, { 8192 }, { 16384 }, { 32768 },
+      { 65536 }, { 131072 }, { 262144 }, { 524288 }, { 1048576 }
+   };
```

Comment:
**suggestion:** use [source include](https://docutils.sourceforge.io/docs/ref/rst/directives.html#toc-entry-39)

Reply:
I didn't know that was possible? @peknis

Reply:
I'll keep it as is. It will leave some artifacts in the actual source code that I'm not completely sold on. Let's postpone this discussion for now.